src/driver_controller.cpp

```diff
@@ -17,6 +19,9 @@ etl::optional<int> handle_command(DriverType type, int arg1, int arg2, int arg3)
             break;
         case DriverType::GET_TIME:
             return timer4_controller.get_time_ns();
+        case DriverType::TERMINAL_OUTPUT:
+            printf((char*)arg1);
```

Comment:
isnt this debug printf? or am I tweaking

Reply:
yea but this effectively prevents it from being preempted by other things _in our OS_. Corruption is caused when printf makes syscalls _to the NRF sdk_ but then _our OS_ triggers an interrupt